openml-datarobot/src/main/java/com/feedzai/openml/datarobot/DataRobotModelCreator.java

```diff
@@ -197,6 +222,24 @@ private String predictor2Str(final Predictor predictor) {
         final SortedSet<String> nominalValues = ((CategoricalValueSchema) schema.getTargetFieldSchema().getValueSchema())
                 .getNominalValues();
 
+        if (IS_BOOLEAN_MODEL.test(targetModelValues)) {
+            if (Objects.equals(
+                    nominalValues.stream().map(StringUtils::lowerCase).collect(Collectors.toSet()),
+                    BOOLEAN_VALUES.stream().map(StringUtils::lowerCase).collect(Collectors.toSet())
+            )) {
+                return new TreeSet<>(BOOLEAN_VALUES);
```

Comment:
this needs to be a SortedSet, or else the assumption done in DataRobotModelCreator#loadModel might fail

`nominalValues.first().equals(targetModelValues[0]),`

Reply:
TreeSet is a sorted set. Otherwise the method would not even compile :)

Reply:
but it won't preserve that the new set was the same order of nominalValues,

`
TreeSet<String> ts1 = new TreeSet<String>(); 
ts1.add("B");
ts1.add("A");
ts1.toString(); // [A, B]
`

Reply:
Clarified that SortedSet is based on a comparator order and not on insert order.